gcm/models.py

```diff
@@ -18,21 +19,57 @@ def get_device_model():
     return load_object(conf.GCM_DEVICE_MODEL)
 
 
-class GCMMessage(api.GCMMessage):
+class GCMMessage(models.Model, api.GCMMessage):
+
+    SENT = "SENT"
+    PENDING = "PENDING"
+    FAILED = "FAILED"
+    NONE = "NONE"
+
+    STATUS_CHOICES = (
+        (SENT, "Sent"),
+        (PENDING, "Pending"),
+        (FAILED, "Failed"),
+        (NONE, "None"))
+
+    topic = models.CharField(
+        verbose_name=_("Topic"), max_length=50, blank=True)
+    created = models.DateTimeField(
+        verbose_name=_("Created"), auto_now_add=True)
+    message = models.TextField(
+        verbose_name=_("Message"))
+    response = models.CharField(
+        verbose_name=_("Response"), max_length=255)
+    status = models.CharField(
+        verbose_name=_("Status"), max_length=16, choices=STATUS_CHOICES, default=NONE)
+    devices = models.ManyToManyField('gcm.Device')
+    #priority = models.CharField(verbose_name=_("Priority"), max_length=255)
+    #content_available = models.CharField(verbose_name=_("Available Content"), max_length=255)
+
     GCM_INVALID_ID_ERRORS = ['InvalidRegistration',
                              'NotRegistered',
                              'MismatchSenderId']
 
-    def send(self, data, registration_ids=None, **kwargs):
+    def send(self, data, registration_ids=None, topic=None, **kwargs):
```

Comment:
We should create an instance of the model, and then call the `send` method without any unnecessary parameters.